keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ArtifactPublisher.kt

```diff
@@ -21,24 +26,63 @@ interface ArtifactPublisher<T : DeliveryArtifact> : SpinnakerExtensionPoint {
   val supportedVersioningStrategies: List<SupportedVersioningStrategy<*>>
 
```

Comment:
I'm conflicting with the name. As this interface is not only responsible for publishing, I think we can rename it to `artifactHandler` or something silimar

Reply:
I went with `ArtifactSupplier` in the end! I think that's better!